src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -573,7 +580,7 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
             // it from the child list so that we don't add it to the deletion
             // list.
             existingChildren.delete(
-              newFiber.key === null ? newFiber.index : newFiber.key
+              newFiber.key === null ? newIdx : newFiber.key
```

Comment:
If we matched on an implicit key, that was `newIdx. `newFiber.index` can be different because it uses the `lastPlacedIndex` counter which is different.